src/hooks/useMessageStore.ts

```diff
@@ -12,7 +13,12 @@ const useMessageStore = () => {
   const [messageStore, dispatchMessages] = useReducer(
     (state: MessageStore, { peerAddress, messages }: MessageStoreEvent) => ({
       ...state,
-      [peerAddress]: [...(state[peerAddress] || []), ...(messages || [])],
+      [peerAddress]: [...(state[peerAddress] || []), ...(messages || [])]
+        .filter(
+          (value, index, self) =>
+            self.findIndex((x) => value.id === x.id) === index
+        )
```

Comment:
Can you leave a small comment explaining what this `filter` is doing? I'm still trying to figure it out 😅

Reply:
Ok I think understand now, having looked properly at the line above. A comment would still be super useful though!

Reply:
done [b97aae3](https://github.com/liteflow-labs/chat/pull/15/commits/b97aae31a07dbb73330edd56e7e316b8f5cf6500)